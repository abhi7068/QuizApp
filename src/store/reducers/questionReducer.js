import * as actionTypes from "../actions/actionTypes";
const initialState = {
    quiz: [],
};
const questionReducer = (state = initialState, action) => {
    console.log(action);
    switch (action.type) {
        case actionTypes.ADD_QUESTION:
            console.log(action)
            return {
                ...state,
              quiz:[...state.quiz,action.question]
            };
        default:
            return state;
    }
};

export default questionReducer;