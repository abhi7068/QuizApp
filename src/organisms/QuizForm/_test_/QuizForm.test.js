import React from 'react'
import { render, cleanup,fireEvent } from  '@testing-library/react';
import '@testing-library/jest-dom';
import { createStore} from 'redux';
import { Provider } from 'react-redux';
import  questionReducer from "../../../store/reducers/questionReducer" 
import { BrowserRouter as Router, withRouter } from 'react-router-dom';
import QuizForm from '../index';
const store=createStore(questionReducer);
afterEach(cleanup);

 
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <QuizForm  />
      </Provider>
    );
    expect(
      asFragment(
        render(
          <Provider store={store}>
            <QuizForm  />
          </Provider>
        )
      )
    ).toMatchSnapshot();
  });
  describe("Testing handler", () => {
    test("test", () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <Router>
                    <QuizForm />
                </Router>
            </Provider>
        );
        const name = getByTestId("Input");
      // const answer = getByTestId("output");
       // const  = getByTestId("name");
       fireEvent.change(name, { target: { value: "welcome" } });
      // fireEvent.change(answer, { target: { value: "welcome1" } });
        //fireEvent.change(name,{ target: { value: "welcome" } });
        expect(getByTestId("Input").value).toEqual("welcome");
      // expect(answer.value).toEqual("welcome1");
    
    });
     
})