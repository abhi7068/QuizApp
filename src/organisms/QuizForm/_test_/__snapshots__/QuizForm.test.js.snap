// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <h2>
    No questions available
  </h2>
</DocumentFragment>
`;
